rust/src/storage/s3/store.rs

```diff
@@ -0,0 +1,734 @@
+use crate::{
+    mask::{Integers, Mask, MaskedModel},
+    model::Model,
+};
+use futures::ready;
+use rusoto_core::{credential::ProvideAwsCredentials, HttpClient, Region, RusotoError};
+use rusoto_s3::{
+    CreateBucketError,
+    CreateBucketOutput,
+    CreateBucketRequest,
+    Delete,
+    DeleteObjectsError,
+    DeleteObjectsOutput,
+    DeleteObjectsRequest,
+    GetObjectError,
+    GetObjectOutput,
+    GetObjectRequest,
+    ListObjectsV2Error,
+    ListObjectsV2Output,
+    ListObjectsV2Request,
+    ObjectIdentifier,
+    PutObjectError,
+    PutObjectOutput,
+    PutObjectRequest,
+    S3Client,
+    StreamingBody,
+    S3,
+};
+use std::{
+    collections::HashMap,
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use thiserror::Error;
+use tokio::{io::AsyncReadExt, stream::Stream};
+
+#[derive(Debug, Error)]
+pub enum StorageError {
+    #[error("S3Error")]
+    S3Error,
+}
+
+#[derive(Clone)]
+struct Buckets(HashMap<&'static str, &'static str>);
```

Comment:
Since this is global and immutable, I think it could even be made `const`. I'm not sure what can be done these days with `const` (there has been a lot of work lately in the compiler in that area), so you might need either [`phf`](https://docs.rs/phf/0.8.0/phf/) or [`lazy_static`](https://docs.rs/lazy_static/1.4.0/lazy_static/index.html), not sure.

Maybe we can keep that for another PR though.

Reply:
Nice! I will take a look